src/Book.java

```diff
@@ -35,4 +41,23 @@ public void printBook() {
         System.out.println('\n' + getAuthorFullName() + '\n' + getName() + '\n'
                 + getManufactureYear() + " года выпуска");
     }
+
+    @Override
+    public boolean equals(Object o) {
+        if (o == null || getClass() != o.getClass()) return false;
+        Book book = (Book) o;
+        return manufactureYear == book.manufactureYear && Objects.equals(author, book.author) && Objects.equals(name, book.name);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(author.getFullName(), name, manufactureYear);
```

Comment:
```suggestion
        return Objects.hash(author, name, manufactureYear);
```
нужно объект автора передавать